tests/rptest/tests/compaction_e2e_test.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import sys
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.default import DefaultClient
+from rptest.services.cluster import cluster
+
+from rptest.services.compacted_verifier import CompactedVerifier, Workload
+
+
+class CompactionE2EIdempotencyTest(RedpandaTest):
+    def __init__(self, test_context):
+        extra_rp_conf = {}
+
+        self.segment_size = 5 * 1024 * 1024 if not self.debug_mode else 1024 * 1024
+        self.partition_count = 3
+
+        super(CompactionE2EIdempotencyTest,
+              self).__init__(test_context=test_context,
+                             extra_rp_conf=extra_rp_conf)
+
+    def topic_segments(self):
+        storage = self.redpanda.node_storage(self.redpanda.nodes[0])
+        topic_partitions = storage.partitions("kafka", self.topic)
+
+        return [len(p.segments) for p in topic_partitions]
+
+    @cluster(num_nodes=4)
+    # sys.maxsize is a special high cardinality case where no keys should be
+    # compacted away. With transactions enabled, all the aborted batches
+    # should be compacted away from the log and should not show up in
+    # consumed list.
```

Comment:
this comment looks obsolete, why we dropped the verification with totally random keys ?

Reply:
you're right, I initially thought that this scenario isn't important but it is resource intensive (all keys are unique so it keeps all written messages in memory twice for produce and consume) so I've dropped. but now I have a second thought and I'll add a specialized workload to cover it